app/tests/routers/test_data_decoder.py

```diff
@@ -71,3 +84,154 @@ async def test_view_data_decoder(self, session: AsyncSession):
             },
         )
         self.assertEqual(response.status_code, 422)
+
+    @database_session
+    async def test_view_data_decoder_with_chain_id(self, session: AsyncSession):
+        example_abi = cast(
+            ABI,
+            [
+                {
+                    "inputs": [
+                        {
+                            "internalType": "uint256",
+                            "name": "droidId",
+                            "type": "uint256",
+                        },
+                        {
+                            "internalType": "uint256",
+                            "name": "numberOfDroids",
+                            "type": "uint256",
+                        },
+                    ],
+                    "name": "buyDroid",
+                    "outputs": [],
+                    "stateMutability": "nonpayable",
+                    "type": "function",
+                },
+            ],
+        )
+
+        example_swapped_abi = cast(
+            ABI,
+            [
+                {
+                    "inputs": [
+                        {
+                            "internalType": "uint256",
+                            "name": "numberOfDroids",
+                            "type": "uint256",
+                        },
+                        {
+                            "internalType": "uint256",
+                            "name": "droidId",
+                            "type": "uint256",
+                        },
+                    ],
+                    "name": "buyDroid",
+                    "outputs": [],
+                    "stateMutability": "nonpayable",
+                    "type": "function",
+                },
+            ],
+        )
```

Comment:
I would preffer ABIS in mock files to keep clean the tests.

Reply:
Agree, I will merge both PRs and then do it, if not I will have to fix conflicts